woocommerce/woocommerce
# Add components to @woocommerce/customer-effort-score

### All Submissions:






### Changes proposed in this Pull Request:



**Note: This PR just moves files around and updates imports. No functional changes or more in-depth refactoring was done. There are a lot of files/lines changed, but the changes should be relatively easy to look through.**

This PR updates the source folder organization of the `@woocommerce/customer-effort-score` and moves the following into this package:

- `useCustomerEffortScoreExitPageTracker`
- `ALLOW_TRACKING_OPTION_NAME`
- `getExitPagePage`
- `addExitPage`
- `removeExitPage`
- `addCustomerEffortScoreExitPageListener`
- `removeCustomerEffortScoreExitPageListener`
- `triggerExitPageSurvey`

The usage of these in `woocommerce/admin/client` have been updated as well.

Note: `ProductMVPFeedbackModal` will be moved from `@woocommerce/customer-effort-score` to `@woocommerce/product-editor` as part of #37131.

Partially addresses #36722.



### How to test the changes in this Pull Request:




Note: make sure tracking is enabled under WooCommerce > Settings > Advanced > WooCommerce.com

1. Load this branch and run `pnpm install && pnpm run build` and make sure everything builds correctly.
2. Smoke test WCAdmin pages and make sure that any CES-related functionality works correctly.
   
    * Import CES snackbar - Go to Products > All Products; click "Import"; without continuing, click back to Products > All Products (see testing instructions on #36656)
    * CES footer on new product editor page - make sure "Share feedback" button brings up feedback modal.

3. Open your console and check if `window.wc.customerEffortScore` exists with the following components:

- `useCustomerEffortScoreExitPageTracker`
- `ALLOW_TRACKING_OPTION_NAME`
- `getExitPagePage`
- `addExitPage`
- `removeExitPage`
- `addCustomerEffortScoreExitPageListener`
- `removeCustomerEffortScoreExitPageListener`
- `triggerExitPageSurvey`



### Other information:




### FOR PR REVIEWER ONLY:
